### ISSUE TITLE
Incorrect or failed rendering of stereochemical or radical structures

### ISSUE THREAD

**Issue by elvati** [Acylation/obsidian-chem#20]
### What happened?

Many radical and stereochemical structures are rendered incorrectly, fail to render, or are represented with some information loss (chirality). A few examples, based on structures taken from the [OPENSMILES specification](http://opensmiles.org/opensmiles.html):

## 1-propyl radicals
```smiles 
CC[CH3]
CCC

CC[CH2]
CCc 

CC[CH1]
CC[CH]

CC[CH0]
CC[C]
```
All these SMILES render identically as propane:
![image](https://github.com/Acylation/obsidian-chem/assets/11050735/3c154121-2513-4b5b-a9be-840886ab5c6e)

## Tetrahedral Centers
```smiles
N[C@](Br)(O)C
# chirality information lost
# 

CCN[C@](Br)(O)C   
# partially rendered as a string, chirality information lost, 
# clipping of the initial H

N[C@](Br)(O)CCC
# partially rendered as a string, chirality information lost
# clipping of the final H subscript

CCN[C@](Br)(O)CCC
# rendered correctly!
```
![image](https://github.com/Acylation/obsidian-chem/assets/11050735/d3d30069-5c01-4964-9787-a0ee65bf3727)

## Tetrahedral Allene-like Systems
```smiles
NC(Br)=[C@]=C(O)C
# does not render
```

### What did you expect to happen?

While a correct representation would be ideal, alternatively the plugin could render problematic or unimplemented SMILES as is, maybe with some graphical indication that the SMILES has not been processed (to avoid potential confusion with SMILES that render as text, like C(I)(Br)(Cl)F ). This change would be particularly important for SMILES that do not render at all, as they would not be visible at all in Live Preview and Reading mode.

### How can we reproduce it (as minimally and precisely as possible)?

To reproduce the issues, copy the above code blocks in Obsidian and compare with the output of other sources (for example, Marvin JS, rdkit).

### Plugin version

0.1.2

### Obsidian version

1.3.5

**Comment by Acylation** [comment]
Thank you for your report! The problems are caused by various reasons, and I’ll address them separately.

1. Radicals
This issue is related to an upstream problem, https://github.com/reymond-group/smilesDrawer/issues/110. Since the smilesDrawer repository is not actively maintained, one solution could be to modify the package directly or use rdkit.js instead. Please note that rdkit.js can only render quoted radical syntax, such as CC[CH2], but not CCc. There are also some differences in handling lowercase letters.

2. Stereochemical Centers
This issue is caused by the compact draw option mentioned in issue #19. A quick fix could be to disable this option by default and allow users to configure it.

![image](https://github.com/Acylation/obsidian-chem/assets/73122375/cf6ae2e9-a80e-4c19-a3f8-16ab8072541d)

![image](https://github.com/Acylation/obsidian-chem/assets/73122375/111a0022-585b-40a3-88be-09e89b9993ef)

3. Allene-like Systems
The smilesDrawer does not recognize these systems. I completely agree that the plugin should catch parsing errors and inform users or display the original strings directly.

**Comment by Acylation** [comment]
Partly solved in v0.2.0
I will try including other cheminfo core then.
